src/app/_components/SearchInput/SearchAutoCompleteDropDown.js

```diff
@@ -11,51 +12,65 @@ export function SearchAutoCompleteDropDown() {
     debouncedQuery,
     autoCompleteItems,
     isAutoCompleteOpen,
+    isInputFocused,
+    setIsAutoCompleteOpen,
     isAutoCompleteLoading,
     navigateToAutoCompleteItem,
   } = useSearchContext();
   const [listOverflown, setListOverflown] = useState(false);
   const onItemsOverflow = useCallback(state => {
     setListOverflown(state);
   }, []);
-  
+  const autoCompleteRef = useRef(null);
+  useClickOutside(autoCompleteRef, () => {
+    if (!isInputFocused) {
```

Comment:
It would be nice to move out callback function from useClickOutside and wrap it with `useCallback` hook for better performance.

Reply:
Moved the handlers into useCallback